spock-mockable/src/main/java/io/github/joke/spockmockable/internal/MockableExtension.java

```diff
@@ -64,35 +66,61 @@ private static void installTransformer() {
                     .withArgs(PROPERTIES_FILE);
         }
     }
 
-    private static Set<String> readClassesFromProperties(final InputStream stream) throws IOException {
+    private static Set<String> extractPackagesFromPropertyResource() {
+        try (final InputStream stream = MockableExtension.class.getResourceAsStream(PROPERTIES_FILE)) {
+            if (stream == null) {
+                log.warn("@Mockable did not find the generated properties file '{}'. Either you did not annotate any tests or the build setup is broken.", PROPERTIES_FILE);
+                return emptySet();
+            }
+
+            return readPropertyFromProperties(stream, "packages");
+        } catch (final IOException e) {
+            throw new ExtensionException("Unable to read properties file '%s' containing mockable package information", e)
+              .withArgs(PROPERTIES_FILE);
+        }
+    }
+
+    private static Set<String> readPropertyFromProperties(final InputStream stream, String key) throws IOException {
         final Properties properties = new Properties();
         properties.load(stream);
 
-        return Stream.of(properties.getProperty("classes", "")
-                        .split(","))
-                .collect(toSet());
+        return Stream.of(properties.getProperty(key, "")
+            .split(","))
+          .collect(toSet());
     }
 
-    private static void buildAndInstallTransformer(final Set<String> classes) {
+    private static void buildAndInstallTransformer(final Set<String> classes, Set<String> packages) {
         new AgentBuilder.Default(new ByteBuddy().with(TypeValidation.DISABLED))
                 .ignore(none())
                 .with(new InstallationListener())
                 .with(new DiscoveryListener())
                 .with(InitializationStrategy.NoOp.INSTANCE)
                 .with(RETRANSFORMATION)
                 .with(REDEFINE)
-                .type(typeDescription -> classes.contains(typeDescription.getName()))
+                .type(typeDescription -> isNotATest(typeDescription) && (isInClasses(classes, typeDescription) || isInPackages(packages, typeDescription)))
                 .transform(MockableExtension::transform)
                 .installOn(instrumentation);
     }
 
+    private static boolean isNotATest(TypeDescription typeDescription) {
+        return !typeDescription.isAssignableTo(Specification.class);
+    }
+
+    private static boolean isInClasses(Set<String> classes, TypeDescription typeDescription) {
+        return classes.contains(typeDescription.getName());
+    }
+
+    private static boolean isInPackages(Set<String> packages, TypeDescription typeDescription) {
+        return packages.stream().anyMatch(packageName -> Optional.ofNullable(typeDescription.getCanonicalName()).map(canonicalName -> canonicalName.startsWith(packageName + ".")).orElse(false));
```

Comment:
split the stream to make it a little more readable.

```
.stream()
.anyMatch()
.map()
.orElse()
```

Reply:
Of course.